skimage/exposure/histogram_matching.py

```diff
@@ -8,17 +8,27 @@ def _match_cumulative_cdf(source, template):
     Return modified source array so that the cumulative density function of
     its values matches the cumulative density function of the template.
     """
-    src_values, src_unique_indices, src_counts = np.unique(source.ravel(),
-                                                           return_inverse=True,
-                                                           return_counts=True)
-    tmpl_values, tmpl_counts = np.unique(template.ravel(), return_counts=True)
+    if source.dtype.kind == 'u':
+        src_lookup = source.ravel()
+        src_counts = np.bincount(src_lookup)
+        tmpl_counts = np.bincount(template.ravel())
+        tmpl_values = np.arange(len(tmpl_counts))
+        
+        tmpl_values_exist = tmpl_counts > 0
+        tmpl_counts = tmpl_counts[tmpl_values_exist]
+        tmpl_values = tmpl_values[tmpl_values_exist]
```

Comment:
The following should be equivalent and avoids the call to `arange`

```suggestion

        # omit values where the count was 0
        tmpl_values = np.nonzero(tmpl_counts)[0]
        tmpl_counts = tmpl_counts[tmpl_values]
```

Reply:
I was matching the style that was used elsewhere in the code. This seems like a separate concern from this pull request, but as far as I am concerned it's OK to use reshape.